src/averageAges.js

```diff
@@ -36,8 +43,20 @@ function calculateMenAverageAge(people, century) {
  *
  * @return {number}
  */
-function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+function calculateWomenAverageAge(people, withChildren = false) {
+  let women = people.filter(person =>
+    person.sex === 'f');
+
+  women = withChildren
+    ? women.filter(woman =>
+      people.filter(person =>
```

Comment:
Instead of nesting filter methods, combine the conditions in a single filter method for better readability.